src/core/cpu_patches.cpp

```diff
@@ -714,6 +846,99 @@ static std::pair<bool, u64> TryPatch(u8* code, PatchModule* module) {
     return std::make_pair(false, instruction.length);
 }
 
+#if defined(ARCH_X86_64)
+// We need to check, before patching, if there's enough space for a relative jump to the trampoline.
+// If there isn't, the instruction must be handled specially in the illegal instruction handler
+// itself.
+static bool ShouldNotBePatched(void* code_address) {
+    ZydisDecodedInstruction instruction;
+    ZydisDecodedOperand operands[ZYDIS_MAX_OPERAND_COUNT];
+    const auto status =
+        Common::Decoder::Instance()->decodeInstruction(instruction, operands, code_address);
```

Comment:
The way things are laid out currently we're decoding the instruction twice, once here and then again in either HandleIllegalInstruction or TryPatch. Could probably refactor this a bit so we only have to decode once.

Reply:
Done